pkg/controller/namespace/deletion/namespaced_resources_deleter.go

```diff
@@ -406,44 +410,46 @@ type gvrDeletionMetadata struct {
-	estimate, err := d.estimateGracefulTermination(gvr, namespace, namespaceDeletedAt)
+	estimate, err := d.estimateGracefulTermination(ctx, gvr, namespace, namespaceDeletedAt)
 	if err != nil {
-		klog.V(5).Infof("namespace controller - deleteAllContentForGroupVersionResource - unable to estimate - namespace: %s, gvr: %v, err: %v", namespace, gvr, err)
+		logger.V(5).Info("Namespace controller - deleteAllContentForGroupVersionResource - unable to estimate", "namespace", namespace, "resource", gvr, "err", err)
 		return gvrDeletionMetadata{}, err
 	}
-	klog.V(5).Infof("namespace controller - deleteAllContentForGroupVersionResource - estimate - namespace: %s, gvr: %v, estimate: %v", namespace, gvr, estimate)
+	logger.V(5).Info("Namespace controller - deleteAllContentForGroupVersionResource - estimate", "namespace", namespace, "resource", gvr, "estimate", estimate)
 
 	// first try to delete the entire collection
-	deleteCollectionSupported, err := d.deleteCollection(gvr, namespace)
+	deleteCollectionSupported, err := d.deleteCollection(ctx, gvr, namespace)
 	if err != nil {
 		return gvrDeletionMetadata{finalizerEstimateSeconds: estimate}, err
 	}
 
 	// delete collection was not supported, so we list and delete each item...
 	if !deleteCollectionSupported {
-		err = d.deleteEachItem(gvr, namespace)
+		err = d.deleteEachItem(ctx, gvr, namespace)
 		if err != nil {
 			return gvrDeletionMetadata{finalizerEstimateSeconds: estimate}, err
 		}
 	}
 
 	// verify there are no more remaining items
 	// it is not an error condition for there to be remaining items if local estimate is non-zero
-	klog.V(5).Infof("namespace controller - deleteAllContentForGroupVersionResource - checking for no more items in namespace: %s, gvr: %v", namespace, gvr)
-	unstructuredList, listSupported, err := d.listCollection(gvr, namespace)
+	logger.V(5).Info("Namespace controller - deleteAllContentForGroupVersionResource - checking for no more items in namespace", "namespace", namespace, "resource", gvr)
+	unstructuredList, listSupported, err := d.listCollection(ctx, gvr, namespace)
 	if err != nil {
-		klog.V(5).Infof("namespace controller - deleteAllContentForGroupVersionResource - error verifying no items in namespace: %s, gvr: %v, err: %v", namespace, gvr, err)
+		logger.V(5).Info("Namespace controller - deleteAllContentForGroupVersionResource - error verifying no items in namespace", "namespace", namespace, "resource", gvr, "err", err)
 		return gvrDeletionMetadata{finalizerEstimateSeconds: estimate}, err
 	}
 	if !listSupported {
 		return gvrDeletionMetadata{finalizerEstimateSeconds: estimate}, nil
 	}
-	klog.V(5).Infof("namespace controller - deleteAllContentForGroupVersionResource - items remaining - namespace: %s, gvr: %v, items: %v", namespace, gvr, len(unstructuredList.Items))
+	logger.V(5).Info("Namespace controller - deleteAllContentForGroupVersionResource - items remaining", "namespace", namespace, "resource", gvr, "items", len(unstructuredList.Items))
```

Comment:
@soltysh I can't see a nil check for the `unstructuredList`. Should we handle this (maybe separate PR?)

Reply:
It cannot be nil. That can only happen when an error is returned by `List` and that is checked.